marconi-core/src/Marconi/Core/Transformer/WithTracer.hs

```diff
@@ -463,3 +477,29 @@ instance
   => Resetable (t m) event (WithTrace m indexer)
   where
   reset = resetVia unwrap
+
+-- | A wrapper to hide the focus of a prism
+data SomePrism e = forall a. SomePrism (APrism' e a)
+
+-- | A helper for logging `IndexerError`s in an `IndexerEvent` trace
+logIndexerError
+  :: (MonadIO m)
+  => Trace m (IndexerEvent point)
+  -> IndexerError
+  -> m ()
+logIndexerError = logError IndexerFailed [SomePrism _StopIndexer]
+
+-- | A helper that allows us to add an exclusion list of error constructors
+logError
+  :: forall m e' e
+   . (MonadIO m)
+  => (e' -> e)
+  -> [SomePrism e']
```

Comment:
While the variable name makes the purpose explicit, can you also add variable documentation here?